src/plugins/spec/spec.c

```diff
@@ -0,0 +1,842 @@
+			{
+				localCh->missing = getConfOption(localConflictMeta);
+			}
+			break;
+		case SET:
+			if((localConflictMeta = (Key *)keyGetMeta(specKey, "conflict/set/member")) != NULL)
+			{
+				localCh->member = getConfOption(localConflictMeta);
+			}
+			else if((localConflictMeta = (Key *)keyGetMeta(specKey, "conflict/set/invalid")) != NULL)
+			{
+				localCh->invalid = getConfOption(localConflictMeta);
+			}
+			else if((localConflictMeta = (Key *)keyGetMeta(specKey, "conflict/set/count")) != NULL)
+			{
+				localCh->count = getConfOption(localConflictMeta);
+			}
+			else if((localConflictMeta = (Key *)keyGetMeta(specKey, "conflict/set/conflict")) != NULL)
+			{
+				localCh->conflict = getConfOption(localConflictMeta);
+			}
+			else if((localConflictMeta = (Key *)keyGetMeta(specKey, "conflict/set/range")) != NULL)
+			{
+				localCh->range = getConfOption(localConflictMeta);
+			}
+			else if((localConflictMeta = (Key *)keyGetMeta(specKey, "conflict/set/missing")) != NULL)
+			{
+				localCh->missing = getConfOption(localConflictMeta);
+			}
+			break;
+	}
+}
+
+static Conflict getConflict(Key *conflictMeta)
+{
+	if(!strcmp(keyName(conflictMeta), "conflict/invalid"))
+		return INVALID;
+	else if(!strcmp(keyName(conflictMeta), "conflict"))
+		return CONFLICT;
+	else if(!strcmp(keyName(conflictMeta), "conflict/hasInvalidMembers"))
+		return ARRAYMEMBER;
+	else if(!strcmp(keyName(conflictMeta), "conflict/range"))
+		return OUTOFRANGE;
+	else if(!strcmp(keyName(conflictMeta), "conflict/invalidSubCount"))
+		return SUBCOUNT;
+	else if(!strcmp(keyName(conflictMeta), "conflict/missing"))
+		return MISSING;
+	else
+		return NAC;
+}
+
+static int handleInvalidConflict(Key *parentKey, Key *key, OnConflict onConflict)
+{
+	int ret = 0;
+	switch(onConflict)
+	{
+		case ERROR:
+			ELEKTRA_SET_ERRORF(142, parentKey, "Invalid key %s\n", keyName(key));
+			ret = -1;
+			break;
+		case WARNING:
+			ELEKTRA_ADD_WARNINGF(143, parentKey, "Invalid key %s\n", keyName(key));
+			break;
+		case INFO:
+			{
+				const char *infoString = "Invalid key ";
+				const size_t len = elektraStrLen(infoString)+elektraStrLen(keyName(key))-1;
+				char *buffer = elektraMalloc(len);
+				snprintf(buffer, len, "Invalid key %s\n", keyName(key));
+				elektraMetaArrayAdd(key, "logs/spec/info", buffer);
+				elektraFree(buffer);
+			}
+			break;
+		case IGNORE:
+
+			break;
+	}
+	return ret;
+}
+
+static int handleArrayConflict(Key *parentKey, Key *key, Key *conflictMeta, OnConflict onConflict)
+{
+	int ret = 0;
+	const char *problemKeys = elektraMetaArrayToString(key, keyName(conflictMeta), ", ");
+	switch(onConflict)
+	{
+		case ERROR:
+			ELEKTRA_SET_ERRORF(142, parentKey, "%s has invalid array key members: %s\n", keyName(key), problemKeys);
+			ret = -1;
+			break;
+		case WARNING:
+			ELEKTRA_ADD_WARNINGF(143, parentKey, "%s has invalid array members: %s\n", keyName(key), problemKeys);
+			break;
+		case INFO:
+			{
+				const char *infoString = "invalid array members: ";
```

Comment:
i've already tried that, because of all the differences in the log entries in each conflict a helper function would make things just unnecessarily complicated.